app/src/main/java/com/keylesspalace/tusky/StatusListActivity.kt

```diff
@@ -221,17 +229,23 @@ class StatusListActivity : BottomSheetActivity(), HasAndroidInjector {
         val tag = hashtag ?: return true
 
         lifecycleScope.launch {
+            var filterCreateSuccess = false
+            val hashedTag = if (tag.startsWith('#')) tag else "#" + tag
+
             mastodonApi.createFilter(
                 title = "#$tag",
                 context = listOf(FilterV1.HOME),
                 filterAction = Filter.Action.WARN.action,
                 expiresInSeconds = null
             ).fold(
                 { filter ->
-                    if (mastodonApi.addFilterKeyword(filterId = filter.id, keyword = tag, wholeWord = true).isSuccess) {
-                        mutedFilter = filter
-                        updateTagMuteState(true)
+                    if (mastodonApi.addFilterKeyword(filterId = filter.id, keyword = hashedTag, wholeWord = true).isSuccess) {
+                        // must be requested again; otherwise does not contain the keyword (but server does)
+                        mutedFilter = mastodonApi.getFilter(filter.id).getOrNull()
```

Comment:
This is a bit odd (third api call).

We could open the filters list any time - as is done as fallback below.

Reply:
Why do you need this?

Reply:
What? This call?
To get the correct current filter with that id.

Otherwise the filter opened will be incomplete (and cannot directly be saved for example).

Reply:
Another observation (which could be done in this PR): This code in EditFilterViewModel and StatusListActivity is double code - esp. for this rather complex "fallback to v1" case. They could be joined.

Reply:
These are the two and a half open questions here:

* Go for a single `createFilter`: this requires testing and/or research why it was "terrible" in the past; and related the question if this is necessary at all here (it is not if always the filter list is displayed; see "FiltersActivity vs. EditFilterActivity" below)
* Join double code (here)?

Reply:
`createFilter`/`updateFilter`: It's like the comment said - it just didn't work. The filter would be created, but the keywords would be missing or not updated.
See the discussion [starting here](https://matrix.to/#/!PWkUGGpeMZWDPmdfRE:matrix.org/$1673017151510578sqjwI:matrix.org?via=m.charlag.space&via=matrix.org&via=tchncs.de)

It's possible that something has been fixed in mastodon since then, I didn't debug the server 🤷

Reply:
Thanks. Can you try it again?

Adding 
```
 @Field("keywords_attributes[][keyword]") keywords: List<String>? = null,
 @Field("keywords_attributes[][whole_word]") wholeWord: List<Boolean>? = null
```

to `createFilter`

and to its call for example in `muteTag`

```
                listOf("#$tag"),
                listOf(true)
```

(Which server are you testing?)

Reply:
At the time, I was testing on my instance (glitch) and also mastodon.gamedev.place (vanilla masto.host mastodon)